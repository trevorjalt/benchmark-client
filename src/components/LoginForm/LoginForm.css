.Login__icon {
    height: 4vh;
    margin-right: 1vw;
}

.Login__button {
    display: flex;
    justify-content: center;
}

.LoginSubmit {
    display: flex;
    align-items: center;
    margin: 4vh 0 4vh 0;

}
