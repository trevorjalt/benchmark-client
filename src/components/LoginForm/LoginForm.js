import React, { Component } from 'react'
import AuthApiService from '../../services/auth-api-service'
import WorkoutContext from '../../contexts/WorkoutContext'
import { Button, Input } from '../Utils/Utils'
import LoginIcon from '../Header/images/login-icon.png'
import './LoginForm.css'

export default class LoginForm extends Component {
    static defaultProps = {
        onLoginSuccess: () => {}
    }

    static contextType = WorkoutContext

    state = { 
        error: null,
    }

    handleSubmitBasicAuth = event => {
        event.preventDefault()
        const { username, user_password } = event.target

        username.value = ''
        user_password.value = ''
        this.props.onLoginSuccess()
    }

    handleSubmitJwtAuth = event => {
        event.preventDefault()
        this.setState({ error: null })

        const { username, user_password } = event.target

        AuthApiService.postLogin({
            username: username.value,
            user_password: user_password.value,
        })
            .then(res => {
                username.value = ''
                user_password.value = ''
                this.props.onLoginSuccess()
            })
            .catch(res => {
                this.setState({ error: res.error })
            })
    }

    render() {
        const { error } = this.state
        return (
            <form
                className='LoginForm'
                onSubmit={this.handleSubmitJwtAuth}
            >
                <div role='alert'>
                    {error && <p className='red'>{error}</p>}
                </div>
                <div className='username' id='username'>
                    <label htmlFor='LoginForm__username'>
                        Username
                    </label>
                    <Input
                        required
                        name='username'
                        id='LoginForm__username'
                        aria-required='true'
                        aria-labelledby='username'
                    >
                    </Input>
                </div>
                <div className='user_password' id='password'>
                    <label htmlFor='LoginForm__password'>
                        Password
                    </label>
                    <Input
                        required
                        name='user_password'
                        type='password'
                        id='LoginForm__password'
                        aria-required='true'
                        aria-labelledby='password'
                    >
                    </Input>
                </div>
                <div className='Login__button'>
                    <Button 
                        className='LoginSubmit'
                        type='submit'
                    >
                        <img src={LoginIcon} alt='Login Button logo' className='Login__icon' />
                        Login
                    </Button>
                </div>
            </form>
        )
    }
}