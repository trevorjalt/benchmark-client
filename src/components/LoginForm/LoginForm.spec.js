import React from 'react'
import LoginForm from './LoginForm'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

describe(`LoginForm Component`, () => {
    it (`renders without crashing`, () => {
        const wrapper = 
            shallow (
                <LoginForm />
            )
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})