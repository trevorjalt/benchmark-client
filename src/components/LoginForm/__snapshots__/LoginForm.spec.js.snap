// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Component renders without crashing 1`] = `
<form
  className="LoginForm"
  onSubmit={[Function]}
>
  <div
    role="alert"
  />
  <div
    className="username"
  >
    <label
      htmlFor="LoginForm__username"
    >
      Username
    </label>
    <Input
      id="LoginForm__username"
      name="username"
      required={true}
    />
  </div>
  <div
    className="user_password"
  >
    <label
      htmlFor="LoginForm__password"
    >
      Password
    </label>
    <Input
      id="LoginForm__password"
      name="user_password"
      required={true}
      type="password"
    />
  </div>
  <div
    className="Login__button"
  >
    <Button
      className="LoginSubmit"
      type="submit"
    >
      <img
        alt="Login"
        className="Login__icon"
        src="login-icon.png"
      />
      Login
    </Button>
  </div>
</form>
`;
